src/main/java/ejercicio/banco/repository/MySqlAccountRepository.java

```diff
@@ -45,53 +45,51 @@
 
     @Override
     public Account find(int id) {
-
         Connection con = dataBaseUtil.startConnection();
 
         try {
-            PreparedStatement pstmt = con.prepareStatement("select * from ACCOUNTS where id=" + id);
+            PreparedStatement pstmt = con.prepareStatement("select * from ACCOUNTS where id = " + id);
             ResultSet rs = pstmt.executeQuery();
-            if(rs.getInt("id")== id) {
+            if (rs.next()) {
                 String name = rs.getString("name");
                 Float money = rs.getFloat("money");
                 String iban = rs.getString("iban");
                 return new Account(id, name, money, iban);
             }
             throw new EntityNotFoundException(String.format("Account with ID = %d does not exist", id));
-    } catch(Exception e)
-
-    { e.printStackTrace();
-    } finally{
-        dataBaseUtil.closeConections(con);
-    }return null;
-}
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            dataBaseUtil.closeConections(con);
+        }
+        return null;
+    }
 
     @Override
     public List<Account> findByName(String name) {
         List<Account> accounts = new ArrayList();
-
         Connection con = dataBaseUtil.startConnection();
 
         try {
             PreparedStatement pstmt = con.prepareStatement("select * from ACCOUNTS where name = " + name);
             ResultSet rs = pstmt.executeQuery();
-            while (rs.next()) {
-                if (rs.getString("name") == name) {
+            if (rs.next()) {
```

Comment:
This is not correct. Either you do `if (rs.next())` when you are expecting only one record or `while (rs.next())` when you are expecting more than one, but not both.

Reply:
At current state I put first the if (rs.next()) followed by the while (rs.next()) because I want to separate the return statement from the thrown Exception (Since I have to use the return outside the while).